lib/web_ui/lib/src/engine/canvaskit/renderer.dart

```diff
@@ -0,0 +1,427 @@
+
+  /// The scene host, where the root canvas and overlay canvases are added to.
+  DomElement? _sceneHost;
+  DomElement? get sceneHost => _sceneHost;
+
+  @visibleForTesting
+  late Rasterizer rasterizer = Rasterizer();
+
+  set resourceCacheMaxBytes(int bytes) => rasterizer.setSkiaResourceCacheMaxBytes(bytes);
+
+  @override
+  Future<void> initialize() async {
+    if (windowFlutterCanvasKit != null) {
+      canvasKit = windowFlutterCanvasKit!;
+    } else if (useH5vccCanvasKit) {
+      if (h5vcc?.canvasKit == null) {
+        throw CanvasKitError('H5vcc CanvasKit implementation not found.');
+      }
+      canvasKit = h5vcc!.canvasKit!;
+      windowFlutterCanvasKit = canvasKit;
+    } else {
+      canvasKit = await downloadCanvasKit();
+      windowFlutterCanvasKit = canvasKit;
+    }
+  }
+
+  String get canvasKitBuildUrl =>
+    configuration.canvasKitBaseUrl + (kProfileMode ? 'profiling/' : '');
+  String get canvasKitJavaScriptBindingsUrl =>
+      '${canvasKitBuildUrl}canvaskit.js';
+  String canvasKitWasmModuleUrl(String canvasKitBase, String file) =>
+      canvasKitBase + file;
+
+  /// Download and initialize the CanvasKit module.
+  ///
+  /// Downloads the CanvasKit JavaScript, then calls `CanvasKitInit` to download
+  /// and intialize the CanvasKit wasm.
+  @visibleForTesting
+  Future<CanvasKit> downloadCanvasKit() async {
+    await _downloadCanvasKitJs();
+    final Completer<CanvasKit> canvasKitInitCompleter = Completer<CanvasKit>();
+    final CanvasKitInitPromise canvasKitInitPromise =
+        CanvasKitInit(CanvasKitInitOptions(
+      locateFile: allowInterop((String file, String unusedBase) =>
+          canvasKitWasmModuleUrl(canvasKitBuildUrl, file)),
+    ));
+    canvasKitInitPromise.then(allowInterop((CanvasKit ck) {
+      canvasKitInitCompleter.complete(ck);
+    }));
+    return canvasKitInitCompleter.future;
+  }
+
+  /// Downloads the CanvasKit JavaScript file at [canvasKitBase].
+  Future<void> _downloadCanvasKitJs() {
+    final String canvasKitJavaScriptUrl = canvasKitJavaScriptBindingsUrl;
+
+    final DomHTMLScriptElement canvasKitScript = createDomHTMLScriptElement();
+    canvasKitScript.src = canvasKitJavaScriptUrl;
+
+    final Completer<void> canvasKitLoadCompleter = Completer<void>();
+    late DomEventListener callback;
+    void loadEventHandler(DomEvent _) {
+      canvasKitLoadCompleter.complete();
+      canvasKitScript.removeEventListener('load', callback);
+    }
+    callback = allowInterop(loadEventHandler);
+    canvasKitScript.addEventListener('load', callback);
+
+    patchCanvasKitModule(canvasKitScript);
+
+    return canvasKitLoadCompleter.future;
+  }
+
+  @override
+  void reset(FlutterViewEmbedder embedder) {
+    /// CanvasKit uses a static scene element that never gets replaced, so it's
+    /// added eagerly during initialization here and never touched, unless the
+    /// system is reset due to hot restart or in a test.
```

Comment:
Inline comments should use `//` instead of `///` (the latter is for dartdocs only)